engine/runtime-integration-tests/src/test/java/org/enso/interpreter/test/ForeignMethodInvokeTest.java

```diff
@@ -141,6 +146,71 @@ public void testParallelInteropWithJavaScript() throws Exception {
     assertEquals(12, res2.getArrayElement(2).asInt());
   }
 
+  @Test
+  public void testParallelAccessToState() throws Exception {
```

Comment:
This is the fist test that checks two threads accessing `State` don't see each other changes.